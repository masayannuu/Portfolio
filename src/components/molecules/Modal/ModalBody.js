import React from 'react'
import PropTypes from 'prop-types'
import { Image } from 'reactbulma'

const ModalBody = ({ imageUrl, description, teamData }) => (
  <div className="content">
    <Image src={imageUrl} />
    <h3>説明</h3>
    <p>{description}</p>
    <h3>チームについて</h3>
    <dl>
      <dt>人数</dt>
      <dd>{teamData.number}人</dd>
      <dt>役割</dt>
      <dd>{teamData.roll}</dd>
      <dt>やったこと</dt>
      <dd>{teamData.work}</dd>
    </dl>
  </div>
)

ModalBody.propTypes = {
  imageUrl: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  teamData: PropTypes.shape({
    number: PropTypes.number,
    roll: PropTypes.string,
    work: PropTypes.string
  }).isRequired
}

export default ModalBody
