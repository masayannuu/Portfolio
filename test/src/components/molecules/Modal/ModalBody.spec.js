import React from 'react'
import Enzyme from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'

import ModalBody from '../../../../../src/components/molecules/Modal/ModalBody'

const description =
  'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字'

const team_data = {
  number: 3,
  roll: '設計・開発主担当',
  work:
    'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ'
}

describe('Modal ModalBody Molecules', () => {
  it('should render ModalBody', () => {
    const wrapper = Enzyme.shallow(<ModalBody
      title="File Transfer Application"
      imageUrl="https://bulma.io/images/placeholders/1280x960.png"
      description={description}
      teamData={team_data}
      />)
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })
})
