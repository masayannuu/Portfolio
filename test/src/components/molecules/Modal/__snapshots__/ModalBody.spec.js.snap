// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal ModalBody Molecules should render ModalBody 1`] = `
<ModalBody__StyledDiv
  className="content"
>
  <Image
    src="https://bulma.io/images/placeholders/1280x960.png"
  />
  <h3>
    説明
  </h3>
  <dl>
    <Paragraph
      text="この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字"
    />
  </dl>
  <h3>
    チームについて
  </h3>
  <dl>
    <ModalBody__StyledDt>
      人数
    </ModalBody__StyledDt>
    <dd>
      3
      人
    </dd>
    <ModalBody__StyledDt>
      役割
    </ModalBody__StyledDt>
    <dd>
      設計・開発主担当
    </dd>
    <ModalBody__StyledDt>
      やったこと
    </ModalBody__StyledDt>
    <Paragraph
      text="この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ"
    />
  </dl>
</ModalBody__StyledDiv>
`;
